File: algorithm/binarysearch/python/BinarySearch.py
def binarySearch(arrs,target):
    low = 0
    high = len(arrs) - 1
    while low <= high:
        mid = (low + high) >> 1
        midval = arrs[mid]
        if midval < target:
            low = mid + 1
        elif midval > target:
            high = mid - 1
        else:
            return mid
        
    return -1

if __name__=='__main__':
    arrs = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610]
    target = 155
    print(binarySearch(arrs,target))

File: algorithm/binarysearch/java/BinarySearch.java
/**
 * BinarySearch
 */
public class BinarySearch {

    public int binarySearch(int[] arrs, int target) {
        int len = arrs.length;
        int low = 0;
        int high = len - 1;
        while (low <= high) {
            int mid = (low + high) >>> 1;
            int midVal = arrs[mid];
            if (midVal < target) {
                low = mid + 1;
            } else if (midVal > target) {
                high = mid - 1;
            } else {
                return mid;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arrs = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610};
        int target = 610;
        BinarySearch bs = new BinarySearch();
        System.out.println(bs.binarySearch(arrs, target));
    }
}